core/backend/src/ECSqlStatement.ts

```diff
@@ -414,6 +416,236 @@ export class ECSqlStatement implements IterableIterator<any>, IDisposable {
   }
 }
 
+/** Executes ECSQL INSERT/UPDATE/DELETE statements.
+ *
+ * A statement must be prepared before it can be executed, and it must be released when no longer needed.
+ * See [ECDb.withCachedWriteStatement]($backend) for a convenient and
+ * reliable way to prepare, execute, and then release a statement.
+ *
+ * A statement may contain parameters that must be filled in before use by the **bind** methods.
+ *
+ * Once prepared (and parameters are bound, if any), the statement is executed by calling [ECSqlStatement.stepForInsert]($backend).
+ *
+ * > Preparing a statement can be time-consuming. The best way to reduce the effect of this overhead is to cache and reuse prepared
+ * > statements. A cached prepared statement may be used in different places in an app, as long as the statement is general enough.
+ * > The key to making this strategy work is to phrase a statement in a general way and use placeholders to represent parameters that will vary on each use.
+ *
+ * See also
+ * - [Executing ECSQL]($docs/learning/backend/ExecutingECSQL) provides more background on ECSQL and an introduction on how to execute ECSQL with the iTwin.js API.
+ * - [Code Examples]($docs/learning/backend/ECSQLCodeExamples) illustrate the use of the iTwin.js API for executing and working with ECSQL
+ * @public
+ */
+export class ECSqlWriteStatement implements IDisposable {
```

Comment:
So this is correct. I was more into `ECDb.ExecuteNoneQueryECSql(ecql: string, args?: ECSqlBinder)` Though I realize `ECSqlBinder` does not let you bind struct since its use in context of Query where we do not allow binding structs/arrays. So what you have here look fine. It requires less code change and can stay to the scope of ECDb.